rttest/src/rttest.cpp

```diff
@@ -415,7 +426,19 @@ int Rttest::init(size_t iterations, struct timespec update_period,
   this->params.sched_priority = sched_priority;
   this->params.stack_size = stack_size;
 
-  this->params.filename = filename;
+  if (filename != nullptr) {
+    size_t n = strlen(filename);
+    this->params.filename = static_cast<char*>(std::malloc(n*sizeof(char) + 1));
+    if (!this->params.filename) {
+      fprintf(stderr, "Failed to allocate major pagefaults buffer\n");
+      return -1;
+    }
+    memset(this->params.filename, 0, n*sizeof(char) + 1);
```

Comment:
indeed. I can get away with only setting index N to 0 (since a string of length N will span indices 0 through N-1 and the null terminator will be at index N).